pkg/services/aci/types.go

```diff
@@ -9,6 +9,45 @@ type provisioningParameters struct {
 	Ports       []int   `json:"ports"`
 }
 
+// GetProvisionParametersSchema generates the schema for instance
+// provisioning parameters
+func GetProvisionParametersSchema() map[string]*service.ParameterSchema {
```

Comment:
Two things about this function...

1. Can we avoid exporting this? As it stands currently, the modules now export nothing except their "constructor" and the functions that the `ServiceModule` or `ServiceManager` interfaces require them to implement. The idea of modules exporting nothing, in my mind, helps keep things nice and tidy.

2. I see that for some modules that expose multiple services (where each service has potentially different parameters), there are multiple functions for returning parameter schemas. I'd like it if, in all cases (e.g. even for modules with a single service manager, we could make these "get schema" helper functions use the appropriate service manager as a receiver. imo, this is another way to keep things tidy and comprehensible, by grouping all functions related to a given service under its service manager.

Reply:
The above is applicable to all modules.

Reply:
👍